service/topology_state_machine.hh

```diff
@@ -83,6 +83,9 @@ struct topology {
     // operation untill the node becomes normal
     std::unordered_map<raft::server_id, request_param> req_param;
 
+    // Supported features, for non-left nodes
+    std::unordered_map<raft::server_id, std::set<sstring>> supported_features;
```

Comment:
Have you considered putting the set inside `replica_state`?

Reply:
No, but makes sense to move it there.